File: 1642-water-bottles/water-bottles.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var numWaterBottles = function(numBottles, numExchange) {
    let totalDrinks = numBottles;
    let emptyBottles = numBottles;

    while (emptyBottles >= numExchange) {
        let newBottles = Math.floor(emptyBottles / numExchange);
        totalDrinks += newBottles;
        emptyBottles = emptyBottles % numExchange + newBottles;
    }

    return totalDrinks;
};

File: 1642-water-bottles/water-bottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int result = numBottles;
        while(numBottles >= numExchange) {
            int canExchange = (int) (numBottles / numExchange);
            result += canExchange;
            numBottles = canExchange + (numBottles % numExchange);
        }
        return result;
    }
}